src/imgbased/plugins/service.py

```diff
@@ -55,9 +54,8 @@ def _generate_iqn(self):
         if os.path.exists(initiator):
             return
         log.debug("Description=Generate a random iSCSI initiator IQN name")
-        suuid = str(uuid.uuid4()).split("-")[-1]
-        factory_f = File("/usr/share/factory/etc/iscsi/initiatorname.iscsi")
-        iqn = factory_f.contents.split(":")[0] + ":" + suuid + "\n"
+        identifier = command.call("/usr/sbin/iscsi-iname")
+        iqn = "InitiatorName=" + identifier.decode("utf-8").strip() + "\n"
```

Comment:
I think that a call to strip() is not required since we already do that as part of the command.call() i.e. note https://github.com/oVirt/imgbased/blob/db2a669d6382211e80c536f6ee618b88bec5fc87/src/imgbased/command.py#L14

But this is not a critical issue.